ecommerce/api.py

```diff
@@ -424,3 +425,30 @@ def create_unfulfilled_order(user):
         redeem_coupon(coupon_version=latest_coupon_version(coupon), order=order)
     order.save_and_log(user)
     return order
+
+
+def get_full_price_coupon_product_set():
+    """
+    Queries the database for CouponPayments that give a 100% off discount and returns those
+    CouponPayments in a tuple with the Product that they apply to.
+
+    Returns:
+        iterable of tuple(CouponPayment, Product): An iterable of CouponPayments paired with the Product that
+            they apply to.
+    """
+    full_coupon_payments = CouponPayment.objects.annotate(
+        max_created_on=Max("versions__created_on")
+    ).filter(
+        versions__coupon_type=CouponPaymentVersion.SINGLE_USE,
+        max_created_on=F("versions__created_on"),
+        versions__amount=1,
+    )
+    for coupon_payment in full_coupon_payments:
+        product_coupons = (
+            CouponEligibility.objects.select_related("product")
+            .filter(coupon__enabled=True, coupon__payment=coupon_payment)
+            .distinct("product")
+        )
+        if product_coupons.count() != 1:
+            continue
+        yield coupon_payment, product_coupons[0].product
```

Comment:
The dropdown list only shows one of the products that a coupon is valid for, there may be more than one.  I made a coupon valid for 3 products, the dropdown showed one of those products but then different emails got sent different products that the coupon applied to:

<img width="508" alt="Screen Shot 2019-04-25 at 8 08 16 AM" src="https://user-images.githubusercontent.com/187676/56734643-6bbb3480-6731-11e9-8aa6-a81fad6a5204.png">

<img width="442" alt="Screen Shot 2019-04-25 at 8 18 02 AM" src="https://user-images.githubusercontent.com/187676/56735157-b12c3180-6732-11e9-9950-8f180a8545c2.png">
<img width="475" alt="Screen Shot 2019-04-25 at 8 03 52 AM" src="https://user-images.githubusercontent.com/187676/56734659-71187f00-6731-11e9-9991-78b4df20b97d.png">
<img width="458" alt="Screen Shot 2019-04-25 at 8 17 54 AM" src="https://user-images.githubusercontent.com/187676/56735165-b6897c00-6732-11e9-9bae-89091a299215.png">


This should probably be changed to show the coupon for every product that it is applicable to, and then emails should get sent for that particular product to all user emails if selected:
```python
for product_coupon in product_coupons:
    yield coupon_payment, product_coupon.product
```

Though maybe no one will ever create a 100% off coupon that applies to multiple products (nothing on the coupon creation form prevents you from doing so however) -  @pdpinch what do you think?

Reply:
PS this is after I removed the `if product_coupons.count() == 0` condition to see what would happen.

Reply:
Our primary use case is where bulk enrollment apply to one product (program, course, or courserun). Can we limit the form to only coupons of that type?

Reply:
Limiting the bulk enrollment to one product also will simplify the mailing: just one one email for each email address uploaded.

Reply:
Yes, it already is here, so good as is @gsidebo!

Reply:
I meant to be more explicit about that in the message and in the code comments. I'll add some clarification